doc/source/geocoding.rst

```diff
@@ -1,17 +1,43 @@
+.. currentmodule:: geopandas
+
+.. ipython:: python
+   :suppress:
+
+   import geopandas as gpd
+
+
 
 Geocoding
 ==========
 
-[TO BE COMPLETED]
+``geopandas`` supports geocoding (i.e., converting place names to
+location on Earth) through `geopy`_, an optional dependency of ``geopandas``.
+The following example shows how to use the `Google geocoding API
+<https://developers.google.com/maps/documentation/geocoding/start>`_ to get the
+locations of boroughs in New York City, and plots those locations along
+with the detailed borough boundary file included within ``geopandas``.
+
+.. _geopy: http://geopy.readthedocs.io/
+
+.. ipython:: python
+
+    boros = gpd.read_file(gpd.datasets.get_path("nybb"))
+    print(boros.BoroName)
+    boro_locations = gpd.tools.geocode(boros.BoroName, provider="google")
+
+    import matplotlib.pyplot as plt
+    fig, ax = plt.subplots()
 
+    boros.to_crs({"init": "epsg:4326"}).plot(ax=ax, color="white", edgecolor="black");
+    boro_locations.plot(ax=ax, color="red");
+    @savefig boro_centers_over_bounds.png
+    plt.show();
```

Comment:
The plt.show is not needed I think, you can put the `@savefig` above the previous plotting line